content/en/continuous_integration/pipelines/teamcity.md

```diff
@@ -0,0 +1,95 @@
+  - link: "/continuous_integration/pipelines"
+    tag: "Documentation"
+    text: "Explore Pipeline Execution Results and Performance"
+  - link: "/continuous_integration/troubleshooting/"
+    tag: "Documentation"
+    text: "Troubleshooting CI"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+## Compatibility
+
+- **Supported TeamCity versions**: 2021.2 or higher
+
+- **Partial retries**: View partially retried pipeline executions
+
+- **Queue time**: View amount of time pipeline jobs sit in the queue before processing
+
+- **Pipeline failure reasons**: Identify pipeline failure reasons from error messages
+
+## Configure the Datadog integration
+
+The integration between [TeamCity][1] and Datadog CI Visibility is provided through a TeamCity plugin.
+To activate the integration:
+1. Download the [Datadog CI Integration plugin][5] on the TeamCity server by going to
+Administration -> Plugins -> Browse Plugin Repository.
+2. For the integration to work properly, a [TeamCity composite build][6] must be added as the last build
+of the build chain (this step can be skipped if there is already one). Namely, this build must have
+a dependency on the current last build of the chain and no other builds depending on it. Build chains not having a
+composite build as their last build are ignored by the plugin.
+3. The following configuration parameters need to be present for the TeamCity projects:
+   * `datadog.ci.api.key`: having as value your [Datadog API Key][2].
+   * `datadog.ci.site`: having as value {{< region-param key="dd_site" code="true" >}}.
+   * `datadog.ci.enabled`: having as value `true` (`false`
+   can be used to disable the plugin for a specific project).
+
+   These configuration parameters can be added either to TeamCity subprojects
+   or to the [TeamCity Root Project][10]. When added to the Root project, they are propagated
+   to all the subprojects. For example, to enable the plugin for all projects, `datadog.ci.enabled` can be added with
+   value `true` to the Root Project. More information on defining configuration parameters
+   are available in the [TeamCity Project Hierarchy][9] docs.
+4. Enable the plugin by restarting the TeamCity server.
+
+## Visualize pipeline data in Datadog
+
+The [Pipelines][3] and [Pipeline Executions][4] pages populate with data after the pipelines finish.
+
+**Note**: The Pipelines page shows data for only the default branch of each repository.
+
+## Troubleshooting
+
+All the logs generated by the Datadog CI Integration plugin are stored inside the `teamcity-server.log` file and can be
+accessed from the TeamCity Server by going to "Administration -> Diagnostic -> Server Logs". If the plugin is not behaving
+correctly, these logs can be checked to get additional context on what might be wrong.
+
+## Git user information
```

Comment:
I think we should give a recommendation on what people should do here, before we tell them what does not work

Reply:
Makes sense, let me add that part here

Reply:
Changed!